crates/sparrow-runtime/src/stores/registry.rs

```diff
@@ -47,32 +47,65 @@ impl ObjectStoreRegistry {
 
     pub async fn upload(
         &self,
-        target_url: ObjectStoreUrl,
-        local_file_path: &path::Path,
+        source_path: &path::Path,
+        destination_url: ObjectStoreUrl,
     ) -> error_stack::Result<(), Error> {
-        let target_path = target_url.path()?;
-        let object_store = self.object_store(&target_url)?;
-        let mut local_file = fs::File::open(local_file_path)
+        let target_path = destination_url.path()?;
+        let object_store = self.object_store(&destination_url)?;
+
+        let upload_error = || Error::UploadingObject {
```

Comment:
@jordanrfrazier I've found this pattern useful. When I have a lot of errors in the same method that need the same error, create a helper closure like this that creates the error, then you can do `change_context_lazy(error_helper_name)`.